src/app.js

```diff
@@ -204,14 +327,223 @@ gpii.app.performQuit = function () {
 gpii.app.exit = function (that) {
     if (that.model.keyedInUserToken) {
         fluid.promise.sequence([
-            gpii.rejectToLog(that.keyOut(that.model.keyedInUserToken), "Couldn't logout current user"),
+            gpii.rejectToLog(that.keyOut(), "Couldn't logout current user"),
             gpii.app.performQuit
         ]);
     } else {
         gpii.app.performQuit();
     }
 };
 
+fluid.registerNamespace("gpii.app.pcp");
+
+/**
+ * Sends a message to the given window
+ *
+ * @param pcpWindow {Object} An Electron `BrowserWindow` object
+ * @param messageChannel {String} The channel to which the message to be sent
+ * @param message {String}
+ */
+gpii.app.pcp.notifyPCPWindow = function (pcpWindow, messageChannel, message) {
+    if (pcpWindow) {
+        pcpWindow.webContents.send(messageChannel, message);
+    }
+};
+
+/**
+ * Creates an Electron `BrowserWindow` that is to be used as the PCP window
+ *
+ * @param width {Number} The desired width of the window
+ * @param height {Number} The desired height of the window
+ * @returns {Object} The created Electron `BrowserWindow`
+ */
+gpii.app.pcp.makePCPWindow = function (width, height) {
+    var screenSize = electron.screen.getPrimaryDisplay().workAreaSize;
+    // TODO Make window size relative to the screen size
+    var pcpWindow = new BrowserWindow({
+        width: width,
+        height: height,
+        show: false,
+        frame: false,
+        fullscreenable: false,
+        resizable: false,
+        alwaysOnTop: true,
+        skipTaskbar: true,
+        x: screenSize.width - width,
+        y: screenSize.height - height,
+
+        // Make the app feel more native
+        backgroundColor: "#71c5ef"
+    });
+
+    pcpWindow.on("blur", function () {
+        pcpWindow.hide();
+    });
+
+    var url = fluid.stringTemplate("file://%dirName/pcp/index.html", {
```

Comment:
i) hoist this out into configuration
ii) expand against fluid.module.terms() rather than dirName as in https://github.com/fluid-project/kettle/blob/master/lib/dataSource-file.js#L105 so that this can read file://%gpii-app/pcp/index.html"